app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt

```diff
@@ -1,65 +1,179 @@
 import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
 import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.home.newconversation.contacts.Contact
+import com.wire.android.ui.home.newconversation.contacts.toContact
+import com.wire.android.ui.home.newconversation.search.ContactSearchResult
+import com.wire.android.ui.home.newconversation.search.SearchPeopleState
+import com.wire.android.ui.home.newconversation.search.SearchResultState
+import com.wire.android.util.flow.SearchQueryStateFlow
+import com.wire.kalium.logic.feature.publicuser.GetAllKnownUsersUseCase
+import com.wire.kalium.logic.feature.publicuser.SearchKnownUsersUseCase
+import com.wire.kalium.logic.feature.publicuser.SearchUserDirectoryUseCase
+import com.wire.kalium.logic.functional.Either
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 @HiltViewModel
 class NewConversationViewModel
 @Inject constructor(
-    private val navigationManager: NavigationManager
+    private val navigationManager: NavigationManager,
+    private val searchKnownUsers: SearchKnownUsersUseCase,
+    private val searchPublicUsers: SearchUserDirectoryUseCase,
+    private val getAllKnownUsersUseCase: GetAllKnownUsersUseCase
 ) : ViewModel() {
 
-    val newConversationState = mutableStateOf(
-        NewConversationState(
-            listOf(
-                Contact(
-                    "This is first contact"
-                ), Contact(
-                    "test2"
-                ), Contact(
-                    "test3"
-                ),
-                Contact(
-                    "test1"
-                ), Contact(
-                    "test2"
-                ), Contact(
-                    "test3"
-                ),
-                Contact(
-                    "test1"
-                ), Contact(
-                    "test2"
-                ), Contact(
-                    "test3"
-                ),
-                Contact(
-                    "test1"
-                ), Contact(
-                    "test2"
-                ), Contact(
-                    "test3"
-                ),
-                Contact(
-                    "test1"
-                ), Contact(
-                    "test2"
-                ), Contact(
-                    "This is last contact"
-                )
+    //TODO: map this value out with the given back-end configuration later on
+    private companion object {
+        const val HARDCODED_TEST_DOMAIN = "staging.zinfra.io"
+    }
```

Comment:
I don't think the app should worry about this that much.

We could have a `SearchForUsersUseCase` that takes a search query and an optional `domain`. If the `domain` is not passed, the use-case can be smart enough to use the domain from `SelfUser`

Reply:
I think we can leave like this for now. But it's way better to leave this logic into `CoreLogic`, than leaving all the different apps to figure it out.

Reply:
true, true there is already a TODO to handle it better, i'll probably get back to you once I work on it :P